### ISSUE TITLE
Not all active interfaces report traffic

### ISSUE THREAD

**Issue by leondz** [mattthias/slurm#23]
Running on ubuntu 1804, I have interface names like enx4865ee110703 . This doesn't even have its link speed recognised, though shorter ones such as lo are monitored fine. The interface appears OK in ifconfig:

```
leon@blade:~$ ifconfig
enx4865ee110703: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.28.9.18  netmask 255.255.254.0  broadcast 10.28.9.255
        inet6 fe80::1947:a4c9:89c:16c7  prefixlen 64  scopeid 0x20<link>
        ether 48:65:ee:11:07:03  txqueuelen 1000  (Ethernet)
        RX packets 543425  bytes 519216430 (519.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 237192  bytes 1083171262 (1.0 GB)
        TX errors 2638  dropped 11 overruns 0  carrier 0  collisions 0
```

Perhaps something to do with the ifname length?

**Comment by mattthias** [comment]
Hi @leondz ,

the interface speed never worked on Linux as far as i remember it only worked on FreeBSD. 
The length of the interface name is most likely the problem. Linux supports up to 16 chars for the name (check for IFNAMSIZ in linux/if.h) but that should match with enx4865ee110703.

I just checked the code and indeed the "if_name" can only hold 10 chars:
https://github.com/mattthias/slurm/blob/d4d13ac43a09b018ed123e24bc44d4ded90188a8/slurm.h#L111 

I'll fix this. 

Btw: Do you use the slurm package from the Ubuntu repository or did you compiled slurm?

Best, 
Matthias

**Comment by leondz** [comment]
Aha, thanks. Great. I took it from the main repo, in this instance, but would compile to get the fix.

**Comment by mattthias** [comment]
Hey @leondz , 

i merged your pull-request  #24 so we can close this issue. Thanks for reporting the bug and fixing it! 
And sorry for the delay ;)

Best wishes,
Matthias
